test/Conversion/HWToLLVM/convert_aggregates.mlir

```diff
@@ -0,0 +1,75 @@
+// RUN: circt-opt %s --convert-hw-to-llvm | FileCheck %s
+
+// CHECK-LABEL: @convertBitcast
+func.func @convertBitcast(%arg0 : i32, %arg1: !hw.array<2xi32>, %arg2: !hw.struct<foo: i32, bar: i32>) {
+
+  // CHECK: %[[ONE1:.*]] = llvm.mlir.constant(1 : i32) : i32
+  // CHECK-NEXT: %[[A1:.*]] = llvm.alloca %[[ONE1]] x i32 {alignment = 4 : i64} : (i32) -> !llvm.ptr<i32>
+  // CHECK-NEXT: llvm.store %2, %[[A1]] : !llvm.ptr<i32>
```

Comment:
It is a bit problematic to use SSA names like `%2` directly in the check expression because they are created by a running counter as opposed to value names like `%arg0` and `%c0_i32` (makes the test more fragile). I'd suggest to either use a wildcard like `{{%.+}}` (the problem here is that the test gets weaker, because a pattern might assign the wrong value and you don't check that with a wildcard) or even better to add lines like `[[SOMENAME:%.+]] = builtin.unrealized_conversion_cast %arg1 : !hw.array<2xi32> to !llvm.array<2 x i32>` and then you can use `[[SOMENAME]]` instead of the `%2`.

Reply:
Thanks, hoped there was a nicer way to do this! I'll get it changed :slightly_smiling_face:

Reply:
There's [one portion in the current test](https://github.com/llvm/circt/blob/ea3027f298a86d969a2413f6e9e86c931b98a3dd/test/Conversion/LLHDToLLVM/convert_aggregates.mlir#L42-L50) where raw SSA names are currently used. Not sure if this is for a reason or if it's something that could do with switching out anyway?

Reply:
Ohh, you mean the one of `%1 = hw.array_slice %arg1[%arg0] : (!hw.array<2xi32>) -> !hw.array<1xi32>`, right? That's an oversight and should not be this way 😅

Reply:
Cool, should be fixed in 8884bd5e8b548de68bc6af9497ba32adeae74169!